lib/Parse/ParseDecl.cpp

```diff
@@ -4422,10 +4422,18 @@ ParserStatus Parser::parseDeclItem(bool &PreviousHadSemi,
   return Result;
 }
 
-bool Parser::parseMemberDeclList(SourceLoc LBLoc, SourceLoc &RBLoc,
-                                 SourceLoc PosBeforeLB,
-                                 Diag<> ErrorDiag,
+bool Parser::parseMemberDeclList(SourceLoc &LBLoc, SourceLoc &RBLoc,
+                                 Diag<> LBraceDiag, Diag<> RBraceDiag,
                                  IterableDeclContext *IDC) {
+  if (parseToken(tok::l_brace, LBLoc, LBraceDiag)) {
+    LBLoc = RBLoc = PreviousLoc;
+
+    // Cache the empty result to prevent delayed parsing.
+    Context.evaluator.cacheOutput(
```

Comment:
Why don't we change the evaluator to handle this rather than poking the result into the cache? This currently depends on the cache not really being a cache, but instead global mutable state.  

The refactoring of `parseMemberDeclList` still seems like a readability win to me though.

Reply:
Although we can use `getBrace().Start == getBrace().End` as the indicator of "missing member brace" for now, I thought it's a little brittle.

I implement it like this because "eager member parsing" is doing `cacheOutput()` as well.
https://github.com/apple/swift/blob/09ea5fd1f7fd61ed2ce8be380f1efc3c120a5ba6/lib/Parse/ParseDecl.cpp#L4441-L4458

Reply:
I see. I guess this doesn't make it worse in that case.